### PR TITLE
added storm wiki frontend with themes

### PR DESCRIPTION
**Pull request by jaigouk**
### why

- wish to choose dark / light themes 
- sometimes, search engine may not return results. and there are search engines that are specialised for specific purpose(arxiv. maybe github in the future). I want to be able to choose them before I send request.
- ollama also can fail. So I want to have an option to "fallback" and continue.

### changes 

- themes: dracula soft dark color and other light and dark themes
- engines: duckduckgo, searxng and arxiv 
- llm: ollama, anthropic
- users can change search engine before triggering search
- users can save primary and fallback llm in settings
- save result files as '*.md'
- add date to to top of the result file
- added arize-phoenix to trace.
- added github ci file to test fallback options for search and llm
- change number of display columns
- pagination in sidebar

### note

because this frontend is using "fallbacks", I thought it would be better to have separate example instead of overwriting the existing demo_light.

### screenshots 

| storm-wiki-dark-theme-setting | storm-wiki-dark-llm-setting |
| --- | --- |
| storm-wiki-dark-search-setting | storm-wiki-dark-article-new |
| storm-wiki-dark-article-list | storm-wiki-dark-create |

dark and light theme
| storm-wiki-dark-article | storm-wiki-solarized-article |
| --- | --- |

### CODE REVIEW

File: frontend/storm_wiki/util/search.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+            return []
+
+    def _search_with_fallback(self, query: str) -> List[Dict[str, Any]]:
+        try:
+            results = self._search(self.primary_engine, query)
+        except Exception as e:
+            logger.warning(
+                f"{self.primary_engine} search failed: {str(e)}. Falling back to {self.fallback_engine}."
+            )
+            if self.fallback_engine:
```
Comment (jaigouk) [inline frontend/storm_wiki/util/search.py:127]: using fallback_engine here if the primary engine fails

**Comment by jaigouk** [comment]
once we have more llms and search engines in lm.py and rm.py, I will update the code to dapt to the changes

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+name: Storm WIKI
```
Comment (jaigouk) [inline .github/workflows/ci.yml:1]: I will remove this file because it is related with contribution policy and details https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/approving-workflow-runs-from-public-forks

File: examples/run_storm_wiki_deepseek.py [hunk 3]
```diff
@@ -23,7 +23,11 @@
 import logging
 from argparse import ArgumentParser
 
-from knowledge_storm import STORMWikiRunnerArguments, STORMWikiRunner, STORMWikiLMConfigs
+from knowledge_storm import (
```
Comment (shaoyijia) [inline examples/run_storm_wiki_deepseek.py:26]: The team is currently reviewing this PR - it may take some time since it's a major enhancement on the demo_light. Thanks for working on it! Could you first revert this change on this file? This file seems only to involve format change due to formatter difference. Ideally, this PR only changes `frontend/storm_wiki` and some other peripheral files (e.g., `.gitignore`).

**Comment by jaigouk** [comment]
i guess my PR is not valid because it is not one of following case. 
```
Pull requests for additional language model support to knowledge_storm/lm.py.
Pull requests for additional retrieval model/search engine support to knowledge_storm/rm.py.
Pull requests for new features to frontend/demo_light to assist other developers.
```
even though i did not closed this, but it seems that there is a rule that closes PRs automatically. 
I checked my forked repo. i even prevented to remove this specific branch and disabled workflow. 
If this PR is closed again automatically, then I will wait until the team opens up the possibility in the future.

FYI @shaoyijia

**Comment by jaigouk** [comment]
@shaoyijia when you have time, could you check the rules that are saved in the official repository? it seems there some rules and my PR is violating some rules.

**Comment by Yucheng-Jiang** [comment]
@jaigouk we have no GitHub actions / rules to auto close PRs configured for this repo.

**Comment by shaoyijia** [comment]
Could you open this PR again? I have time today to work on merging it, though I don't have clue why it's automatically closed.

**Comment by jaigouk** [comment]
@shaoyijia @Yucheng-Jiang 
I guess this is from my side. I mirrored my storm repo from my gitea instance. 
I will clean things up and open up a PR again in the weekend. sorry for bothering you with this.
current changes include "category" concept. https://github.com/jaigouk/storm_wiki
because i wanted to group articles in a folder.
